### PR TITLE
Siva | BAH-1146 | Changes for form2 encounter validations

### PR DESCRIPTION
**Pull request by sivareddytw**
- Added new control properties (mandatory, allow future date, multi select and add more)

### CODE REVIEW

File: form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java [hunk 1]
```diff
@@ -96,4 +101,39 @@ private String getFormName(List<String> orderedPathToControl) {
         return !isEmpty(orderedPathToControl) ? orderedPathToControl.get(0) : "";
     }
 
+    @Override
+    public boolean isMultiSelectObs(List<String> orderedControlNames) {
+        final String formName = getFormName(orderedControlNames);
+        if (!formNamesToFormFieldPathMap.containsKey(formName)) {
+            initializeControlFormFieldPaths(formName);
+        }
+        return formNamesToFormControlMap.get(formName).get(orderedControlNames.toString()).getProperties().isMultiSelect();
```
Comment (tarunkumar-tw) [inline form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java:110]: @sivareddytw Did we make sure that all the .get will return not null values.

File: form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java [hunk 2]
```diff
@@ -52,21 +54,24 @@ private void initializeControlFormFieldPaths(String formName) {
 
         final String initialFormFieldPath = initializeFormFieldPath(formName, formLatestVersion);
         final Map<String, String> controlToFormFieldPath = new HashMap<>();
+        final Map<String, Control> formControls = new HashMap<>();
         formNamesToFormFieldPathMap.put(formName, controlToFormFieldPath);
+        formNamesToFormControlMap.put(formName,formControls);
```
Comment (tarunkumar-tw) [inline form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java:59]: @sivareddytw Correct the indentation - Add space after ,

**Review by supriyatw — APPROVED** [review]

File: form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java (hunk 2 above)
Comment (sivareddytw) [reply form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java]: updated..

File: form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java (hunk 1 above)
Comment (sivareddytw) [reply form2-utils/src/main/java/org/bahmni/form2/service/impl/FormFieldPathServiceImpl.java:110]: @tarunkumar-tw method isValidCSVHeader() ensures no NPE

**Review by tarunkumar-tw — APPROVED** [review]

**Review by binduak — APPROVED** [review]

**Pull request merged**